### PR TITLE
847 licence info

### LINKED ISSUE #847: Add Licenses page
It would be good to have a page that lists the licenses for the packages used by fitbenchmarking. This may be informative to users and might help in the future if we wish to distribute a docker container.

### PR DESCRIPTION
**Pull request by tyronerees**
#### Description of Work

Fixes #847

I've changed the links to use `__` at the end, not `_`, to avoid duplicated labels warnings.

My editor has also formatted headings differently (I don't know why!) -- let me know if there is an issue and I can revert.

#### Testing Instructions

1. Check the links work and are formatted correctly
2. Is this the best way to document this information?
3. Are there any licensed plugins that we are missing?

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: docs/source/users/options/jacobian_option.rst [hunk 1]
```diff
@@ -37,6 +37,8 @@ options are:
 
 Default is ``2-point``
 
+**Licence** SciPy is available under a `3-clause BSD Licence <https://github.com/scipy/scipy/blob/master/LICENSE.txt>`__.  Individual packages many have their own (compatible) licences, as listed `here <https://github.com/scipy/scipy/blob/master/LICENSES_bundled.txt>`__.
```
Comment (AndrewLister-STFC) [inline docs/source/users/options/jacobian_option.rst:40]: Not sure if "Many individual..." or my suggestion is correct?
```suggestion
**Licence** SciPy is available under a `3-clause BSD Licence <https://github.com/scipy/scipy/blob/master/LICENSE.txt>`__.  Individual packages may have their own (compatible) licences, as listed `here <https://github.com/scipy/scipy/blob/master/LICENSES_bundled.txt>`__.
```

File: docs/source/users/options/minimizer_option.rst [hunk 2]
```diff
@@ -14,26 +14,28 @@ software.
...
-* `Differential Evolution <https://bumps.readthedocs.io/en/latest/guide/optimizer.html#differential-evolution>`_ (:code:`de`)
+* `Differential Evolution <https://bumps.readthedocs.io/en/latest/guide/optimizer.html#differential-evolution>`__ (:code:`de`)
 
-* `MINPACK <https://github.com/bumps/bumps/blob/96b5100fc3d5b9485bd4a444c83a33617b74aa9d/bumps/mpfit.py>`_ (:code:`mp`)  This is a translation of `MINPACK` to Python.
+* `MINPACK <https://github.com/bumps/bumps/blob/96b5100fc3d5b9485bd4a444c83a33617b74aa9d/bumps/mpfit.py>`__ (:code:`mp`)  This is a translation of `MINPACK` to Python.
 
-
-**Links** `GitHub - bumps <https://github.com/bumps/bumps>`_
+**Licence** The main licence file for Bumps is `here <https://github.com/bumps/bumps/blob/master/LICENSE.txt>`__.  Individual files have their own copyright and licence
+-- please check before use.
```
Comment (AndrewLister-STFC) [inline docs/source/users/options/minimizer_option.rst:36]: I'm not sure this is quite the right wording? Using this option is fine, it's just using it in commercial(?) software or bundling it that would be a problem.

File: docs/source/users/options/minimizer_option.rst [hunk 3]
```diff
@@ -394,24 +418,26 @@ The RALFit minimizers are set as follows:
...
+* `BFGS algorithm <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-bfgs.html>`__ (:code:`BFGS`)
+* `Newton-CG algorithm <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-newtoncg.html>`__  (:code:`Newton-CG`)
+* `L-BFGS-B algorithm <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-lbfgsb.html>`__ (:code:`L-BFGS-B`)
+* `Truncated Newton (TNC) algorithm <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-tnc.html>`__ (:code:`TNC`)
+* `Sequential Least SQuares Programming <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-slsqp.html>`__ (:code:`SLSQP`)
+
+**Links** `Github - SciPy minimize <https://github.com/scipy/scipy/blob/master/scipy/optimize/_minimize.py>`__
 
-**Links** `Github - SciPy minimize <https://github.com/scipy/scipy/blob/master/scipy/optimize/_minimize.py>`_
+**Licence** Scipy is available under a `3-clause BSD Licence <https://github.com/scipy/scipy/blob/master/LICENSE.txt>`__.  Individual packages many have their own (compatible) licences, as listed `here <https://github.com/scipy/scipy/blob/master/LICENSES_bundled.txt>`__.
```
Comment (AndrewLister-STFC) [inline docs/source/users/options/minimizer_option.rst:440]:
```suggestion
**Licence** Scipy is available under a `3-clause BSD Licence <https://github.com/scipy/scipy/blob/master/LICENSE.txt>`__.  Individual packages may have their own (compatible) licences, as listed `here <https://github.com/scipy/scipy/blob/master/LICENSES_bundled.txt>`__.
```

File: docs/source/users/options/minimizer_option.rst [hunk 4]
```diff
@@ -451,18 +479,20 @@ The SciPy least squares minimizers are set as follows:
...
+* `Differential Evolution (derivative-free) <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html#scipy.optimize.differential_evolution>`__ (:code:`differential_evolution`)
+* `Simplicial Homology Global Optimization (SHGO) <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.shgo.html#scipy.optimize.shgo>`__ (:code:`shgo`)
+* `Dual Annealing <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.dual_annealing.html#scipy.optimize.dual_annealing>`__ (:code:`dual_annealing`)
+
+**Links** `Github - SciPy optimization <https://github.com/scipy/scipy/blob/master/scipy/optimize/>`__
 
-**Links** `Github - SciPy optimization <https://github.com/scipy/scipy/blob/master/scipy/optimize/>`_
+**Licence** Scipy is available under a `3-clause BSD Licence <https://github.com/scipy/scipy/blob/master/LICENSE.txt>`__.  Individual packages many have their own (compatible) licences, as listed `here <https://github.com/scipy/scipy/blob/master/LICENSES_bundled.txt>`__.
```
Comment (AndrewLister-STFC) [inline docs/source/users/options/minimizer_option.rst:495]:
```suggestion
**Licence** Scipy is available under a `3-clause BSD Licence <https://github.com/scipy/scipy/blob/master/LICENSE.txt>`__.  Individual packages may have their own (compatible) licences, as listed `here <https://github.com/scipy/scipy/blob/master/LICENSES_bundled.txt>`__.
```

File: docs/source/users/problem_definition_files/cutest.rst [hunk 5]
```diff
@@ -16,6 +16,15 @@ The SIF file format is very powerful, and CUTEst will work with arbitrary
 variable names, however for FitBenchmarking, these must match a set of expected
 variable names.
 
+**Licence** This file format needs PyCUTEst and the pacakges ARCHDefs, CUTEst and
```
Comment (AndrewLister-STFC) [inline docs/source/users/problem_definition_files/cutest.rst:19]:
```suggestion
**Licence** This file format needs PyCUTEst and the packages ARCHDefs, CUTEst and
```

File: docs/source/users/options/minimizer_option.rst [hunk 6]
```diff
@@ -52,20 +54,22 @@ The Bumps minimizers are set as follows:
...
 FitBenchmarking currently supports the DFO minimizers:
 
-* `Derivative-Free Optimizer for Least Squares <http://people.maths.ox.ac.uk/robertsl/dfols/userguide.html>`_ (:code:`dfols`)
+* `Derivative-Free Optimizer for Least Squares <http://people.maths.ox.ac.uk/robertsl/dfols/userguide.html>`__ (:code:`dfols`)
 
-* `Derivative-Free Gauss-Newton Solver <http://people.maths.ox.ac.uk/robertsl/dfogn/userguide.html>`_ (:code:`dfogn`)
+* `Derivative-Free Gauss-Newton Solver <http://people.maths.ox.ac.uk/robertsl/dfogn/userguide.html>`__ (:code:`dfogn`)
 
- **Links** `GitHub - DFO-GN <https://github.com/numericalalgorithmsgroup/dfogn>`_ `GitHub - DFO-LS <https://github.com/numericalalgorithmsgroup/dfols>`_
+**Licence** Both `DFO-GN <https://github.com/numericalalgorithmsgroup/dfogn/blob/master/LICENSE.txt>`__ and `DFO-LS <https://github.com/numericalalgorithmsgroup/dfols/blob/master/LICENSE.txt>`__ are available under the GPL-3 licence.  A proprietary licence is also available from `NAG <`https://www.nag.com/content/worldwide-contact-information>`__ .
```
Comment (AndrewLister-STFC) [inline docs/source/users/options/minimizer_option.rst:70]: This link is broken (I think this is all it needs?)
```suggestion
**Licence** Both `DFO-GN <https://github.com/numericalalgorithmsgroup/dfogn/blob/master/LICENSE.txt>`__ and `DFO-LS <https://github.com/numericalalgorithmsgroup/dfols/blob/master/LICENSE.txt>`__ are available under the GPL-3 licence.  A proprietary licence is also available from `NAG <https://www.nag.com/content/worldwide-contact-information>`__ . 
```

File: docs/source/users/options/minimizer_option.rst [hunk 7]
```diff
@@ -343,14 +363,16 @@ The `matlab_stats` minimizer is set as follows:
...
 
-CERN developed the `Minuit <http://seal.web.cern.ch/seal/snapshot/work-packages/mathlibs/minuit/>`_ package to find the minimum value of a multi-parameter function, and also to compute the uncertainties.
-We interface via the python interface `iminuit <https://iminuit.readthedocs.io>`_ with support for the 2.x series. 
+* `Minuit's MIGRAD <https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.pdf>`__ (:code:`minuit`)
 
-* `Minuit's MIGRAD <https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.pdf>`_ (:code:`minuit`)
+**Links** `Github - iminuit <https://github.com/scikit-hep/iminuit>`__
 
-**Links** `Github - iminuit <https://github.com/scikit-hep/iminuit>`_
+**Licence** iminuit is released under the `MIT licence <https://github.com/scikit-hep/iminuit/blob/develop/LICENSE>`__, while Minuit is `LGPL v2 <http://seal.web.cern.ch/seal/main/license.html>`__ .
```
Comment (AndrewLister-STFC) [inline docs/source/users/options/minimizer_option.rst:375]: I get a forbidden message when I try to follow this Minuit link.

**Review by AndrewLister-STFC — CHANGES_REQUESTED** [review]
Looks pretty good. I think this is a nice low profile way to add this information. Once we have a more standardised approach to class attributes we can look to add something there too.

One thing I did spot that I can't comment on inline is that the MatLab Cureve Fitting Toolbox escaped all of the edits (licence and heading level), so needs to be updated.

File: docs/source/users/options/minimizer_option.rst (hunk 2 above)
Comment (tyronerees) [reply docs/source/users/options/minimizer_option.rst]: Thanks -- I meant this as a warning if people developers want to use bumps in their own software, but can see how that's not at all clear.  I've reworded now -- hopefully that's clearer.

File: docs/source/users/options/minimizer_option.rst (hunk 7 above)
Comment (tyronerees) [reply docs/source/users/options/minimizer_option.rst]: Updated the seal links to root to fix this

**Review by AndrewLister-STFC — APPROVED** [review]

**Pull request merged**